File: solutions/python/largest-series-product/1/largest_series_product.py
def largest_product(series, size):
    """
    series: a sequence of adjacent digits contained within the input
    size: how many digits each series is
    """
    if size > len(series):
        raise ValueError("span must not exceed string length")
    if size < 0:
        raise ValueError("span must not be negative")
    if any(not c.isdigit() for c in series):
        raise ValueError("digits input must only contain digits")
    
    sub_series = [series[i:i+size] for i in range(len(series)) if i + size <= len(series)]
    products = []
    for s in sub_series:
        digits = [int(c) for c in s]
        product = digits[0]
        for i in range(1, len(digits)):
            product *= digits[i]
        products.append(product)
    return max(products)

File: solutions/csharp/largest-series-product/1/LargestSeriesProduct.cs
public static class LargestSeriesProduct {
  public static long GetLargestProduct(string digits, int span) {
    if (span < 0 || span > digits.Length || !digits.All(char.IsDigit)) {
      throw new ArgumentException();
    }
    else if (span == 0) {
      return 1;
    }

    return Enumerable.Range(0, digits.Length - span + 1)
      .Max(i => digits.Substring(i, span)
        .Select(c => int.Parse(c.ToString()))
        .Aggregate(1, (acc, c) => acc * c));
  }
}